quantecon/dle.py

```diff
@@ -0,0 +1,327 @@
+
+    """
+
+    def __init__(self, information, technology, preferences):
+
+        # === Unpack the tuples which define information, technology and preferences === #
+        self.a22, self.c2, self.ub, self.ud = information
+        self.phic, self.phig, self.phii, self.gamma, self.deltak, self.thetak = technology
+        self.beta, self.llambda, self.pih, self.deltah, self.thetah = preferences
+
+        # === Computation of the dimension of the structural parameter matrices === #
+        self.nb, self.nh = self.llambda.shape
+        self.nd, self.nc = self.phic.shape
+        self.nz, self.nw = self.c2.shape
+        junk, self.ng = self.phig.shape
+        self.nk, self.ni = self.thetak.shape
+
+        # === Creation of various useful matrices === #
+        uc = np.hstack((np.eye(self.nc), np.zeros((self.nc, self.ng))))
+        ug = np.hstack((np.zeros((self.ng, self.nc)), np.eye(self.ng)))
+        phiin = np.linalg.inv(np.hstack((self.phic, self.phig)))
+        phiinc = uc.dot(phiin)
+        phiing = ug.dot(phiin)
+        b11 = - self.thetah.dot(phiinc).dot(self.phii)
+        a1 = self.thetah.dot(phiinc).dot(self.gamma)
+        a12 = np.vstack((self.thetah.dot(phiinc).dot(
+            self.ud), np.zeros((self.nk, self.nz))))
+
+        # === Creation of the A Matrix for the state transition of the LQ problem === #
+
+        a11 = np.vstack((np.hstack((self.deltah, a1)), np.hstack(
+            (np.zeros((self.nk, self.nh)), self.deltak))))
+        self.A = np.vstack((np.hstack((a11, a12)), np.hstack(
+            (np.zeros((self.nz, self.nk + self.nh)), self.a22))))
+
+        # === Creation of the B Matrix for the state transition of the LQ problem === #
+
+        b1 = np.vstack((b11, self.thetak))
+        self.B = np.vstack((b1, np.zeros((self.nz, self.ni))))
+
+        # === Creation of the C Matrix for the state transition of the LQ problem === #
+
+        self.C = np.vstack((np.zeros((self.nk + self.nh, self.nw)), self.c2))
+
+        # === Define R,W and Q for the payoff function of the LQ problem === #
+
+        self.H = np.hstack((self.llambda, self.pih.dot(uc).dot(phiin).dot(self.gamma), self.pih.dot(
+            uc).dot(phiin).dot(self.ud) - self.ub, -self.pih.dot(uc).dot(phiin).dot(self.phii)))
+        self.G = ug.dot(phiin).dot(
+            np.hstack((np.zeros((self.nd, self.nh)), self.gamma, self.ud, -self.phii)))
+        self.S = (self.G.T.dot(self.G) + self.H.T.dot(self.H)) / 2
+
+        self.nx = self.nh + self.nk + self.nz
+        self.n = self.ni + self.nh + self.nk + self.nz
+
+        self.R = self.S[0:self.nx, 0:self.nx]
+        self.W = self.S[self.nx:self.n, 0:self.nx]
+        self.Q = self.S[self.nx:self.n, self.nx:self.n]
+
+        # === Use quantecon's LQ code to solve our LQ problem === #
+
+        lq = LQ(self.Q, self.R, self.A, self.B,
+                self.C, N=self.W, beta=self.beta)
+
+        self.P, self.F, self.d = lq.stationary_values()
```

Comment:
@thomassargent30 this will default to using `doubling` as the method for computing. Should we allow the pass through of the method of computation to be `doubling` and `qz`?